### PR TITLE
[BIOMAGE-1209] - Fix Matomo Heatmap

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Link to issue 

https://biomage.atlassian.net/browse/BIOMAGE-1209

#### Link to staging deployment URL

PR branch - https://ui-matomo-staging.scp-staging.biomage.net/

#### Links to any Pull Requests related to this

N.A.

#### Anything else the reviewers should know about the changes here

- Rendering of heatmaps in production is better now.
- Heatmap and session tracking is trigerred manually to provide better heatmap result

To try this out :
- Go to biomage.matomo.cloud. Login using the credentials posted in #business channel.
- Go to "Heatmaps". Choose a heatmap that you'd like to see.
- The pages now render better.

image

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [X] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [X] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/pages/experiments/[experimentId]/plots-and-tables/volcano/index.jsx [hunk 1]
```diff
@@ -71,6 +72,10 @@ const VolcanoPlot = ({ experimentId }) => {
     setDataPointStatus();
   }, []);
 
+  useEffect(() => {
+    if (config) captureNewPageView();
+  }, [config]);
```
Comment (aerlaut) [inline src/pages/experiments/[experimentId]/plots-and-tables/volcano/index.jsx]: Should not be dependent on config change

File: src/components/data-exploration/generic-gene-table/GeneTable.jsx [hunk 2]
```diff
@@ -42,6 +43,7 @@ const GeneTable = (props) => {
 
   useEffect(() => {
```
Comment (cosa65) [inline src/components/data-exploration/generic-gene-table/GeneTable.jsx]: What about having a custom useEffect hook that handles only the matomo  tracking? To split it up from the rest of the page logic.

File: src/components/data-processing/StatusIndicator.jsx [hunk 3]
```diff
@@ -28,9 +28,15 @@ const StatusIndicator = (props) => {
     status: { pipeline },
   } = useSelector((state) => state.experimentSettings.backendStatus);
 
+  const mockPipelineData = {
+    startDate: '2021-01-01T00:00:00',
+    stopDate: '2021-01-01T00:00:00',
+    status: pipelineStatus.SUCCEEDED,
+  };
+
```
Comment (aerlaut) [inline src/components/data-processing/StatusIndicator.jsx]: Remove snapshot mocks

**Comment by aerlaut** [comment]
Changes were made in this PR to trigger heatmap visits manually. However, after testing, it was found that manually triggering the routes, especially if the trigger is placed in a child element, causes multiple calls to Matomo. Further testing shows that triggering when the URL changes is more reliable. Therefore changes in this PR is not merged into master. 

However, the goals of the ticket this PR is meant to address was achieved, which is to have a better snapshot than what was previously taken automatically in Matomo.

**Review by cosa65 — APPROVED** [review]

**Pull request merged**